src/libmongoc/doc/mongoc_bulk_operation_execute.rst

```diff
@@ -25,7 +25,7 @@ Parameters
 ----------
 
 * ``bulk``: A :symbol:`mongoc_bulk_operation_t`.
-* ``reply``: An uninitialized :symbol:`bson:bson_t`.
+* ``reply``: Overwritable storage for a new :symbol:`bson_t`, typically on the stack. See :doc:`BSON out parameters <bson:lifetimes>`.
```

Comment:
I recommend using a substitution for this text, since it is often repeated:

```rst
* ``reply``: A |bson_t-storage-ptr| to contain the results.
```

and add the substitution to the `rst_prolog` in `libmongoc/doc/conf.py`:

```rst
.. |bson_t-storage-ptr| replace::

    non-\ ``NULL`` pointer to :doc:`overwritable storage <bson:lifetimes>` for a :symbol:`bson_t`

.. |bson_t-opt-storage-ptr| replace::

    maybe-\ ``NULL`` pointer to :doc:`overwritable storage <bson:lifetimes>` for a :symbol:`bson_t`
```

specifically as a noun-phrase so that it can be used in place of a type.